unstructured/ingest/interfaces.py

```diff
@@ -11,12 +12,41 @@
 from unstructured.staging.base import convert_to_dict
 
 
+@dataclass
+class StandardConnectorConfig:
+    """Common set of config options passed to all connectors."""
+
+    # where raw documents are stored for processing, and then removed if not preserve_downloads
+    download_dir: str
+    # where to write structured data outputs
+    output_dir: str
+    download_only: bool = False
+    fields_include: str = "element_id,text,type,metadata"
+    flatten_metadata: bool = False
+    metadata_exclude: Optional[str] = None
+    metadata_include: Optional[str] = None
+    partition_by_api: bool = False
+    partition_endpoint: str = "https://api.unstructured.io/general/v0/general"
+    preserve_downloads: bool = False
+    re_download: bool = False
+
+
+class BaseConnectorConfig(ABC):
+    """Abstract definition on which to define connector-specific attributes."""
+
+
+@dataclass
```

Comment:
we make the git connectors dataclasses. to respect that (and assuming others might do the same in the future) we can make the base a dataclass so that way standard_config doesn't need to be redefined on every child (otherwise the init function won't have it as an argument).

Reply:
alternatively we could just let the git connectors redefine it and not make the base a dataclass